### ISSUE TITLE
Wrong os_sorted sorting with special character in filename

### ISSUE THREAD

**Issue by toroConverter** [SethMMorton/natsort#145]
**Describe the bug**
It can happen that the `os_sorted` functionality does not sort correctly files with special character inside

**Expected behavior**
File sorting equal to Windows Explorer

**Environment (please complete the following information):**
 - Python Version: 3.9
 - Natsort Version: 8.0.2
 - OS Windows
 - If the bug involves `LOCALE` or `humansorted`:
    - Is `PyICU` installed? No
    - Do you have a locale set? If so, to what? Italian

**To Reproduce**

```
from natsort import os_sorted

file_list = ['Try.Me.Bug - 09 - One.Two.Three.[text].mkv',
             'Try.Me.Bug - 07 - One.Two.5.[text].mkv',
             'Try.Me.Bug - 08 - One.Two.Three[text].mkv']

file_list2 = ['TryMe - 02 - One Two Three [text].mkv',
              'TryMe_-_03_-_One_Two_Three_[text].mkv',
              'TryMe_-_01_-_One_Two_Three_[text].mkv']

for file in os_sorted(file_list):
    print(file)

for file in os_sorted(file_list2):
    print(file)
```

**Expected sorting:**

file_list
Try.Me.Bug - 07 - One.Two.5.[text].mkv
Try.Me.Bug - 08 - One.Two.Three[text].mkv
Try.Me.Bug - 09 - One.Two.Three.[text].mkv

file_list2
TryMe\_-\_01\_-\_One\_Two\_Three\_[text].mkv
TryMe - 02 - One Two Three [text].mkv
TryMe\_-\_03_-\_One\_Two\_Three\_[text].mkv

**Actual sorting**

file_list
Try.Me.Bug - 08 - One.Two.Three[text].mkv
Try.Me.Bug - 09 - One.Two.Three.[text].mkv
Try.Me.Bug - 07 - One.Two.5.[text].mkv

file_list2
TryMe - 02 - One Two Three [text].mkv
TryMe\_-\_01\_-\_One\_Two\_Three\_[text].mkv
TryMe\_-\_03\_-\_One\_Two\_Three\_[text].mkv

------ 

At the moment the only way to overcome this issue is to use `os_sorted` along with `key=lambda x: re.sub(r'[^a-zA-Z0-9]+', ' ', x)`

**Comment by SethMMorton** [comment]
I agree that the `Try.Me.Bug - 07 - One.Two.5.[text].mkv` case was a bug (at least undesired behavior) and I have a fix proposed in #146.

However, the `TryMe_-_01_-_One_Two_Three_[text].mkv` case is not a bug, and you would need to use something like `key=lambda x: x.replace("_", " ")` to sort it properly. This is because the code has no way to tell that you want to treat `"_"` and `" "` the same unless you tell it to.

**Comment by toroConverter** [comment]
@SethMMorton ok thanks a lot for your feedback and your work!
